### ISSUE TITLE
custom macros not executed in YAML

### ISSUE THREAD

**Issue by defionscode** [architect/parser#35]
**Describe the issue**
Given `Architect 6.0.28`, a custom macro, and a `arc.yaml` file, the macro is not invoked

**Steps to reproduce**

1. Create a `arc.yaml` with something like

    ```yaml
    macros:
    - test-macro
    ```

1. Create the macro in `src/macros/test-macro.js`
1. The macro code for reproduction purposes can just be
    ```js
    module.exports = function testMacro (arc, cloudformation, stage) {
      console.log('test macro')
      return cloudformation
    }
    ```
1. Run `arc deploy`

**Expected behavior**
I would expect to see `test macro` in stdout when running arc deploy. This does not happen with a YAML config, only with a `.arc` style config.

**Comment by brianleroux** [comment]
Sorry accidentally closed (on my phone travelling rn) 

Will have a look today/ think the 2.x parser fixes both these issues but haven't integrated fully yet

**Comment by ryanblock** [comment]
FYI: as of last weekend I certified Parser 2.1 for production release; Architect will be rolling this out in the next week (as soon as I can find the time to update all our packages).

**Comment by ryanblock** [comment]
Confirmed this is fixed in `6.2.5`!
